### PR TITLE
backwards-compatible fix for python3 API changes

### PR DESCRIPTION
**Pull request by v4hn**
This fixes the remaining issues coming up when generating https://github.com/ros-planning/moveit_tutorials .

It would be nice to see a melodic release for these fixes.

### CODE REVIEW

File: src/rosdoc_lite/msgenator.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def _generate_raw_text(raw_text):
 
 def _generate_msg_text_from_spec(package, spec, msg_context, buff=None, indent=0):
     if buff is None:
-        buff = cStringIO.StringIO()
+        buff = io.StringIO()
```
Comment (tfoote) [inline src/rosdoc_lite/msgenator.py]: For full compatibility we have a conditional import to do this: http://wiki.ros.org/python_2_and_3_compatible_code

File: src/rosdoc_lite/landing_page.py [hunk 2]
```diff
@@ -82,7 +82,12 @@ def generate_links(package, manifest, rd_configs):
     :param rd_configs: [dict] package manifest rosdoc configs
     :returns: [str] list of html snippets
     """
-    config_list = [c for c in rd_configs.itervalues() if c['builder'] != 'rosmake']
+    try:
+        rd_configs_values = rd_configs.values()
```
Comment (tfoote) [inline src/rosdoc_lite/landing_page.py]: I believe that we should be able to just use `values()` in both 2.7 and 3.x

http://wiki.ros.org/python_2_and_3_compatible_code

File: src/rosdoc_lite/rdcore.py [hunk 3]
```diff
@@ -150,6 +150,12 @@ def instantiate_template(tmpl, tempvars):
     """
     looks up file within rosdoc_lite package, return its content, may sys.exit on error
     """
-    for k, v in tempvars.iteritems():
+    try:
+        items = tempvars.items()
```
Comment (tfoote) [inline src/rosdoc_lite/rdcore.py]: I believe that we should be able to just use `items()` in both 2.7 and 3.x

http://wiki.ros.org/python_2_and_3_compatible_code

**Comment by v4hn** [comment]
Thanks for the review @tfoote, I addressed your feedback.
Would be nice to see this merged soon.

**Pull request merged**